crates/pallet-domains/src/staking.rs

```diff
@@ -621,10 +667,17 @@ pub(crate) fn do_switch_operator_domain<T: Config>(
         note_pending_staking_operation::<T>(operator.current_domain_id)?;
 
         ensure!(
-            operator.status == OperatorStatus::Registered,
+            *operator.status::<T>(operator_id) == OperatorStatus::Registered,
             Error::OperatorNotRegistered
         );
 
+        // Reject switching domain if there is unconfirmed ER submitted by this operator
```

Comment:
IIUC to be able to switch the operator has to basically go offline (stop producing bundles, but still stake) for a whole challenge period, and then come online, submit switch extrinsic and go offline again until it's processed?
If so, that is quite subpar. I imagine, ideally, we would accept the switch extrinsic, "freeze" the operator stake on the current domain, so they don't produce new bundles and don't reduce other operators' chances and transfer the stake after their last block is out of challenge period.
This said, it can be handled in a separate issue.